src/include/platform/internal/GenericPlatformManagerImpl_POSIX.cpp

```diff
@@ -257,22 +257,65 @@ CHIP_ERROR GenericPlatformManagerImpl_POSIX<ImplClass>::_StartEventLoopTask()
 }
 
 template <class ImplClass>
-CHIP_ERROR GenericPlatformManagerImpl_POSIX<ImplClass>::_Shutdown()
+CHIP_ERROR GenericPlatformManagerImpl_POSIX<ImplClass>::_StopEventLoopTask()
 {
     int err = 0;
+
+    _LockChipStack();
```

Comment:
Why not call the outer version (`LockChipStack()` ?) Why call the inner version?

Reply:
I'm not sure as to the right guideline here - is calling the outer version the more 'right' approach?

Reply:
Fixed.